src/ui/branches_tab.rs

```diff
@@ -689,9 +694,10 @@ impl Component for BranchesTab<'_> {
             return Ok(ComponentInputResult::Handled);
         }
 
-        if let Event::Key(key) = event
-            && key.kind == KeyEventKind::Press
-        {
+        if let Event::Key(key) = event {
+            if key.kind == KeyEventKind::Press {
+                return Ok(ComponentInputResult::Handled);
```

Comment:
This doesn't match the login that was there previously

Reply:
Oops, I think this should be:

```diff
-if key.kind == KeyEventKind::Press
+if key.kind != KeyEventKind::Press
```

The rest of your comments seems fine, can you check again?

 I chose to do early return here to avoid pushing the whole block one indent further, but I can do that if you don't like the return.

Reply:
Yep you're right, the others are right.

Reply:
Yes you're right, the other instances are correct. Can you push a fix for for this and we can merge/

Reply:
Done in ba622a83c3e052faf8453090b8050da004d62b41